packages/datagateway-download/src/downloadApiHooks.ts

```diff
@@ -0,0 +1,231 @@
+  const settings = React.useContext(DownloadSettingsContext);
+  const { facilityName, downloadApiUrl } = settings;
+  return useQuery(
+    'cart',
+    () =>
+      fetchDownloadCart({
+        facilityName,
+        downloadApiUrl,
+      }),
+    {
+      onError: (error) => {
+        handleICATError(error);
+      },
+      staleTime: 0,
+    }
+  );
+};
+
+export const useRemoveAllFromCart = (): UseMutationResult<
+  void,
+  AxiosError,
+  void
+> => {
+  const queryClient = useQueryClient();
+  const settings = React.useContext(DownloadSettingsContext);
+  const { facilityName, downloadApiUrl } = settings;
+
+  return useMutation(
+    () => removeAllDownloadCartItems({ facilityName, downloadApiUrl }),
+    {
+      onSuccess: (data) => {
+        queryClient.setQueryData('cart', []);
+      },
+      onError: (error) => {
+        handleICATError(error);
+      },
+    }
+  );
+};
+
+export const useRemoveEntityFromCart = (): UseMutationResult<
+  DownloadCartItem[],
+  AxiosError,
+  { entityId: number; entityType: 'investigation' | 'dataset' | 'datafile' }
+> => {
+  const queryClient = useQueryClient();
+  const settings = React.useContext(DownloadSettingsContext);
+  const { facilityName, downloadApiUrl } = settings;
+
+  return useMutation(
+    ({ entityId, entityType }) =>
+      removeFromCart(entityType, [entityId], {
+        facilityName,
+        downloadApiUrl,
+      }),
+    {
+      onSuccess: (data) => {
+        queryClient.setQueryData('cart', data);
+      },
+      onError: (error) => {
+        handleICATError(error);
+      },
+    }
+  );
+};
+
+export const useIsTwoLevel = (): UseQueryResult<boolean, AxiosError> => {
+  const settings = React.useContext(DownloadSettingsContext);
+  const { idsUrl } = settings;
+  return useQuery('isTwoLevel', () => getIsTwoLevel({ idsUrl }), {
+    onError: (error) => {
+      handleICATError(error);
+    },
+    staleTime: Infinity,
+  });
+};
+
+// TODO: refactor rest of dg-download to use react-query
+// export const useSubmitCart = (): UseMutationResult<
+//   number,
+//   AxiosError,
+//   {
+//     transport: string;
+//     emailAddress: string;
+//     fileName: string;
+//     zipType?: 'ZIP' | 'ZIP_AND_COMPRESS';
+//   }
+// > => {
+//   const queryClient = useQueryClient();
+//   const settings = React.useContext(DownloadSettingsContext);
+//   const { facilityName, downloadApiUrl } = settings;
+
+//   return useMutation(
+//     ({ transport, emailAddress, fileName, zipType }) =>
+//       submitCart(
+//         transport,
+//         emailAddress,
+//         fileName,
+//         {
+//           facilityName,
+//           downloadApiUrl,
+//         },
+//         zipType
+//       ),
+//     {
+//       onSuccess: (data) => {
+//         queryClient.setQueryData('cart', data);
+//       },
+//       onError: (error) => {
+//         handleICATError(error);
+//       },
+//     }
+//   );
+// };
```

Comment:
Is there a plan to tackle this at some point?

Reply:
I'll likely make it a separate issue once this is merged in - I didn't want to do all that work in this one PR which is already changing a lot